ddtrace/internal/sampling.py

```diff
@@ -154,7 +154,7 @@ def _sample(self, span):
         elif self._sample_rate == 0:
             return False
 
-        return ((span.span_id * KNUTH_FACTOR) % MAX_SPAN_ID) <= self._sampling_id_threshold
+        return ((span._span_id_64bits * KNUTH_FACTOR) % MAX_UINT_64BITS) <= self._sampling_id_threshold
```

Comment:
maybe for a follow-up: if this function is called frequently perhaps we can remove the properties and reduce the number of nested calls for a bit of extra performance